### PR TITLE
PP 10733 add concourse metrics on the deploy to staging pipeline adminusers

### PR DESCRIPTION
**Pull request by Rkotyank**
Added tasks to get metrics in the deploy-to-staging Adminusers in the following jobs:

- deploy-adminusers-to-staging
- smoke-test-adminusers-on-staging
- adminusers-pact-tag
- push-adminusers-to-production-ecr

### CODE REVIEW

**Review by jfharden — COMMENTED** [review]
The ci/tasks/parse-staging-release-tag.yml and ci/scripts/parse-staging-release-tag.js files don't seem to be used? Could they be removed or maybe is something that uses them missing from the PR?

**Comment by Rkotyank** [comment]
> The ci/tasks/parse-staging-release-tag.yml and ci/scripts/parse-staging-release-tag.js files don't seem to be used? Could they be removed or maybe is something that uses them missing from the PR?

yes, that was my doubt as well. I don't think we are using it anywhere in this PR or will need to use it for the other apps. we can remove them I think.

**Comment by Rkotyank** [comment]
> > The ci/tasks/parse-staging-release-tag.yml and ci/scripts/parse-staging-release-tag.js files don't seem to be used? Could they be removed or maybe is something that uses them missing from the PR?
> 
> yes, that was my doubt as well. I don't think we are using it anywhere in this PR or will need to use it for the other apps. we can remove them I think.

@katstevens what is your suggestion?

**Comment by katstevens** [comment]
So we need those scripts because the `application_image_tag` value that we're currently passing through to the metrics task is a string, not a numerical value. We need to slice off the `-release` before we can send it to Hosted Graphite.

**Comment by Rkotyank** [comment]
> So we need those scripts because the `application_image_tag` value that we're currently passing through to the metrics task is a string, not a numerical value. We need to slice off the `-release` before we can send it to Hosted Graphite.

hmm, that's probably why I'm not seeing the metrics in HG. let me see where I need to add that. Thanks Kat

File: ci/pipelines/deploy-to-staging.yml [hunk 1]
```diff
@@ -1845,8 +1852,19 @@ jobs:
         trigger: true
         passed: [deploy-adminusers-to-staging]
       - get: pay-ci
+      - task: parse-staging-release-tag
+        file: pay-ci/ci/tasks/parse-staging-release-tag.yml
+        input_mapping:
+          git-release: endtoend-git-release
```
Comment (katstevens) [inline ci/pipelines/deploy-to-staging.yml:1858]: So these two lines are where the problem is. The  `parse-staging-release-tag.yml` task is currently expecting `inputs` of two folders: `pay-ci` and `ecr-repo`. That's not what we're providing it here 😸

Comment (Rkotyank) [reply ci/pipelines/deploy-to-staging.yml:1858]: @katstevens I made this change
 ```
task: parse-staging-release-tag
        file: pay-ci/ci/tasks/parse-staging-release-tag.yml
        input_mapping:
          ecr-repo: adminusers-ecr-registry-staging
```

with a new error 

```
selected worker: ip-10-1-10-50
An error occurred: ReferenceError: release is not defined
```

File: ci/tasks/parse-staging-release-tag.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# This task parses the staging release candidate tag from an ecr resource and turns it into a *-staging release tag
+#
+# File written:
+#   parse-staging-release-tag/tag : The staging release tag (e.g. 123-release)
+#
```
Comment (katstevens) [inline ci/tasks/parse-staging-release-tag.yml:5]: This comment isn't quite describing what's happening in the task, and we should probably update it to reflect the different reason for its use. 

The old task we copied this code from ([parse-perf-release-tag.yml](https://github.com/alphagov/pay-ci/blob/master/ci/tasks/parse-perf-release-tag.yml)) was designed to take a tag like `123-release` as an input, and output a different kind of tag: `123-perf`. Here, we're taking a tag `123-release`, and outputting the release number, `123`. Let's change the explanatory sentence above and the example of what's written to the output file.

**Comment by katstevens** [comment]
Overall this is looking good - I have requested a change to a comment on one of the files though.

**Pull request merged**